muspinsim/input/input.py

```diff
@@ -117,31 +121,49 @@ def __init__(self, fs=None):
                 pass
 
             # Now parse
+            errors_found = []
             for header, block in raw_blocks.items():
-
-                hsplit = header.split()
-                name = hsplit[0]
-                args = hsplit[1:]
-
                 try:
-                    KWClass = InputKeywords[name]
-                except KeyError:
-                    raise MuSpinInputError(
-                        "Invalid keyword " "{0} ".format(name) + "found in input file"
-                    )
+                    hsplit = header.split()
+                    name = hsplit[0]
+                    args = hsplit[1:]
 
-                if issubclass(KWClass, MuSpinEvaluateKeyword):
-                    kw = KWClass(block, args=args, variables=self._variables)
-                else:
-                    kw = KWClass(block, args=args)
+                    try:
+                        KWClass = InputKeywords[name]
+                    except KeyError:
+                        raise MuSpinInputError(
+                            "Invalid keyword {0} found in input file".format(name)
+                        )
+
+                    if issubclass(KWClass, MuSpinEvaluateKeyword):
+                        kw = KWClass(block, args=args, variables=self._variables)
+                    else:
+                        kw = KWClass(block, args=args)
 
-                kwid = kw.id
+                    kwid = kw.id
 
-                if name != kwid:
-                    self._keywords[name] = self._keywords.get(name, {})
-                    self._keywords[name][kwid] = kw
-                else:
-                    self._keywords[name] = kw
+                    if name != kwid:
+                        self._keywords[name] = self._keywords.get(name, {})
+                        self._keywords[name][kwid] = kw
+                    else:
+                        self._keywords[name] = kw
+                except LarkExpressionError as e:
+                    errors_found += [
+                        "Error occurred when parsing keyword {0}"
+                        "(block starting at line {1})\n{2}".format(
+                            name, block_line_nums[header], str(e)
+                        )
+                    ]
+
+            if errors_found:
+                err_str = ""
+                for i, err in enumerate(errors_found):
+                    err_str += "Error {0}:\n{1}\n\n".format(i, err)
```

Comment:
This output is much better now. I think you could actually drop the "Error N" line, or not have the newline after it (e.g. `Error 1: Error when parsing field X...`), so it doesn't take up so much vertical space.

Printing the total number of errors either before or after the error list would be nice though.

Oh, and we should probably start counting at 1 if we do keep it. 
```suggestion
                    err_str += "Error {0}: {1}\n\n".format(i+1, err)
```